### PR TITLE
Add cache_file parameter for DynamicClient

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

- Add logic for cache file name generation
- Add cache_file when DynamicClient is created

Old PR: https://github.com/ansible-collections/community.kubernetes/pull/405

### CODE REVIEW

File: plugins/module_utils/common.py [hunk 1]
```diff
@@ -181,8 +216,16 @@ def auth_set(*names):
         client = get_api_client._pool[digest]
         return client
 
+    def generate_cache_file(kubeclient):
+        import hashlib
```
Comment (Akasurde) [inline plugins/module_utils/common.py:220]: Let us import this with rest of the imports.

File: plugins/module_utils/common.py [hunk 2]
```diff
@@ -181,8 +216,16 @@ def auth_set(*names):
...
+        cache_file_name = 'k8srcp-{0}.json'.format(hashlib.sha1(get_default_cache_id(kubeclient)).hexdigest())
```
Comment (Akasurde) [inline plugins/module_utils/common.py:221]:
```suggestion
        cache_file_name = 'k8srcp-{0}.json'.format(hashlib.sha256(get_default_cache_id(kubeclient)).hexdigest())
```

Comment (Akasurde) [reply plugins/module_utils/common.py:221]: sha256 is more secure.

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: plugins/module_utils/common.py (hunk 1 above)
Comment (alinabuzachis) [reply plugins/module_utils/common.py]: Moved where all the imports are done.

File: plugins/module_utils/common.py (hunk 2 above)
Comment (alinabuzachis) [reply plugins/module_utils/common.py]: Thank you for reviewing @Akasurde. Done.

**Comment by Akasurde** [comment]
@fabianvf @geerlingguy @gravesm Can you please review this? Thanks in advance.

**Review by gravesm — APPROVED** [review]
lgtm

**Comment by Akasurde** [comment]
@fabianvf @geerlingguy Could you please take a look and let us know if it works for you? Thanks.

**Review by fabianvf — APPROVED** [review]
/lgtm

**Review by Akasurde — APPROVED** [review]

**Comment by Akasurde** [comment]
@alinabuzachis Thanks for the contribution. @gravesm @fabianvf Thanks for the review.

**Pull request merged**